daffodil-core/src/main/scala/edu/illinois/ncsa/daffodil/dsom/PropProviders.scala

```diff
@@ -120,6 +120,7 @@ trait LeafPropProvider
 final class ChainPropProvider(leafProvidersArg: Seq[LeafPropProvider], forAnnotation: String)
   extends Logging with PropTypes {
 
+  override def toString() = Misc.getNameFromClass(this) + "(" + forAnnotation + ")"
   /**
    * for debug/test only
    */
```

Comment:
Consider removing things that are for debug/test only - see if there are actually tests using them.